src/compile/mark/init.ts

```diff
@@ -25,6 +25,10 @@ export function normalizeMarkDef(mark: Mark | MarkDef, encoding: Encoding<string
     markDef.opacity = opacity(markDef.type, encoding);
```

Comment:
Does this also work if we just return undefined from `opacity(..)` when the value is 1? If not directly, we could implement the same approach as for filled below and only set the opacity when it's defined. 

The problem with your code is that undefined does not get set in json but it does get set in objects.

Reply:
I updated the code to delete the opacity property in the case where `specifiedOpacity` equals one. I believe this addresses the concern regarding the difference between undefined in objects and JSON.

I tried an implementation where `opacity(...)` returned undefined when the mark's opacity is one but it seemed wrong since the purpose of `opacity(...)` is for the cases where the opacity property isn't explicitly defined.